apps/ledger-live-mobile/src/newArch/features/FirmwareUpdate/utils/navigateToFirmwareUpdateFlow.tsx

```diff
@@ -0,0 +1,53 @@
+import { ParamListBase, RouteProp } from "@react-navigation/native";
+import { DeviceModelInfo } from "@ledgerhq/types-live";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { ScreenName, NavigatorName } from "~/const";
+import { lastConnectedDeviceSelector } from "~/reducers/settings";
+import { isNewFirmwareUpdateUxSupported } from "./isFirmwareUpdateSupported";
+import { FirmwareUpdateContextEntity } from "@ledgerhq/live-common/device-core/managerApi/entities/FirmwareUpdateContextEntity";
+import { UpdateStep } from "~/screens/FirmwareUpdate";
+
+export function navigateToFirmwareUpdateFlow({
+  lastConnectedDevice,
+  lastSeenDeviceModelInfo: lastSeenDeviceModelInfo,
+  latestFirmware,
+  route,
+  navigation,
+  onBackFromUpdate,
+}: {
+  lastConnectedDevice: ReturnType<typeof lastConnectedDeviceSelector>;
```

Comment:
For the type as 'big' as this one, maybe give it name is a good choice and make it easier to read. 🥹